### PR TITLE
Allow user to provide their own digest algorithm, remove sha3 dependency

### LINKED ISSUE #20: Let user provide their own digest algorithm for verifications
All of the common digest algorithms including `sha3_256` implement the `digest::Digest` trait, so the digest verification function can be made generic with respect to that, and be used for any digest algorithm rather than a pre-defined one.

### PR DESCRIPTION
**Pull request by dralley**
Let user use whichever implementation of `digest::Digest` they like.

closes #20

### CODE REVIEW

**Comment by dralley** [comment]
Since the `verify` feature would no longer require a heavy dependency (`digest::Digest` is just a trait) maybe it would make sense to have `verify` be provided by default?

File: examples/download.rs [hunk 1]
```diff
@@ -72,18 +76,25 @@ fn main() {
         .unwrap();
 
     let dl = downloader::Download::new(
-        "https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-10.7.0-amd64-netinst.iso",
```
Comment (dralley) [inline examples/download.rs:75]: `debian-cd/current/` only keeps the latest version, so the URLs go out of date.  They don't seem to archive the netinstaller versions and the ISOs are too large, so I just went looking for a medium-sized test file that would be stable and Hetzner seems to provide one.

Comment (hunger) [reply examples/download.rs:75]: That is perfectly fine! Thank you!

**Review by hunger — APPROVED** [review]
Thank you for your contribution!

This is a ABI break, isen't it? I think we can get away with it at this time:-)

**Comment by hunger** [comment]
Could you do a quick cargo fmt? Or shall I?

I use [pre-commit](https://pre-commit.com/) to make sure to not forget to clean up when committing:-)

**Comment by dralley** [comment]
It's an API break yes.  Because the old verification function went away

**Comment by dralley** [comment]
I ran `cargo fmt`

**Comment by hunger** [comment]
Thanks! Let's see what the Ci says now:-)

**Pull request merged**